curta/src/chip/register/bit.rs

```diff
@@ -92,3 +92,27 @@ impl<B: Builder> Mul<B> for BitRegister {
         builder.expression(self.expr() * rhs.expr())
     }
 }
+
+impl<B: Builder> Not<B> for BitRegister {
+    type Output = Self;
+
+    fn not(self, builder: &mut B) -> Self::Output {
+        builder.expression(self.not_expr())
+    }
+}
+
+impl<B: Builder> Or<B> for BitRegister {
+    type Output = Self;
+
+    fn or(self, rhs: Self, builder: &mut B) -> Self::Output {
+        builder.expression(self.expr() + rhs.expr() - self.expr() * rhs.expr())
+    }
+}
+
+impl<B: Builder> And<B> for BitRegister {
+    type Output = Self;
+
+    fn and(self, rhs: Self, builder: &mut B) -> Self::Output {
+        builder.expression(self.expr() * rhs.expr())
```

Comment:
Could you change to builder.mul() I feel like it would be good in case the mul implementation is more efficient than the expression one (right now it's the same).